addons/stock/models/stock_move.py

```diff
@@ -620,24 +620,22 @@ def write(self, vals):
         if 'product_uom' in vals and any(move.state == 'done' for move in self):
             raise UserError(_('You cannot change the UoM for a stock move that has been set to \'Done\'.'))
         if 'product_uom_qty' in vals:
-            move_to_unreserve = self.env['stock.move']
```

Comment:
This seems to me a little bit to complicated. I would patch `_recompute_state` to leave in draft state moves being draft if they have 0 quantity registered (check `test_create_move_line_reserved`)

Reply:
> This seems to me a little bit to complicated. I would patch `_recompute_state` to leave in draft state moves being draft if they have 0 quantity registered (check `test_create_move_line_reserved`)

I agree with you.

Reply:
updated